### ISSUE TITLE
Performance problem

### ISSUE THREAD

**Issue by ivanfemia** [abap2xlsx/abap2xlsx#50]
Hi guys,

I have a problem with generating big sheet. I tried to create a file with more than  10000 rows, but it took very long time.
I did some analysis and found problem in method SET_CELL of class ZCL_EXCEL_WORKSHEET. There is following code:
    APPEND ls_sheet_content TO sheet_content.
    SORT sheet_content BY cell_row cell_col
This will sort sheet_content every time a new cell is added. Use of sorted table and INSERT .. INTO TABLE should be faster.

**Comment by ivanfemia** [comment]
This piece of code was removed some times ago due performance problem.
SORT could be performed as I remember only once in the write method...

I will have a look.

**Comment by ivanfemia** [comment]
Fixed in latest commit #79
Ludek  could you please verify performance using latest daily build.

**Comment by ivanfemia** [comment]
There is a problem in ZCL_EXCEL_WORKSHEET~SET_MERGE. A short dump ITAB_ILLEGAL_SORT_ORDER appears on command 
APPEND ls_sheet_content TO sheet_content_merge.

**Comment by ivanfemia** [comment]
Hi Ludek, 
please open a new issue for this problem. I think it is not related to the performance issue. Have you been able to test is successfully?
Best regards
Gregor

**Comment by ivanfemia** [comment]
Hi Gregor,
this problem is caused by change of standard table to sorted table. This change was done due to this issue, so I don't think, it's necessary to open a new issue.
Ludek

**Comment by ivanfemia** [comment]
OK, got it.

**Comment by ivanfemia** [comment]
Hi Ludek,

in our demo reports this is issue is not thrown. 
I suppose that this dump appears only when more than one merged cell is in a sheet, could you please provide me a demo report (e.g. extend the _ZDEMO_EXCEL13_ or create a new report based on ZDEMO_EXCEL13) to test the issue?
I will look to this tonight.

Thanks,
Ivan

**Comment by ivanfemia** [comment]
Ludek,

issue solved, I will commit new code tonight.
If you need a solution in the meantime put into method ZCL_EXCEL_WORKSHEET~SET_MERGE
INSERT ls_sheet_content INTO TABLE sheet_content_merge.
instead of 
APPEND ls_sheet_content TO sheet_content_merge.

Cheers,
Ivan

**Comment by ivanfemia** [comment]
Hi Ludek,

fix committed.

Cheers

**Comment by ivanfemia** [comment]
Hi Ivan,

it's better now, thanks.
